app/main.py

```diff
@@ -2,0 +1,33 @@
+class Animal:
+
+    alive = []
+
+    def __init__(
+        self, name: str, health: int = 100, hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
```

Comment:
The `isinstance` check for `herbivore` being an instance of `Herbivore` is redundant here because the method signature already specifies that `herbivore` should be of type `Herbivore`. You can safely remove this check.